File: src/main/java/com/canefe/story/StoryPlaceholderExpansion.java
package com.canefe.story;
import me.clip.placeholderapi.expansion.PlaceholderExpansion;
import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class StoryPlaceholderExpansion extends PlaceholderExpansion {

    private final Story plugin;

    public StoryPlaceholderExpansion(Story plugin) {
        this.plugin = plugin;
    }

    @Override
    public String getAuthor() {
        return "canefe";
    }

    @Override
    public String getIdentifier() {
        return "story";
    }

    @Override
    public String getVersion() {
        return "1.0.0";
    }

    @Override
    public boolean persist() {
        return true;
    }

    @Override
    public String onRequest(OfflinePlayer player, @NotNull String params) {
        if (player == null) {
            return "";
        }

        if (params.equalsIgnoreCase("quest_title")) {
            if (player.isOnline()) {
                return plugin.getQuestTitle(player.getPlayer());
            }
            return "";
        }

        if (params.equalsIgnoreCase("quest_objective")) {
            if (player.isOnline()) {
                return plugin.getQuestObj(player.getPlayer());
            }
            return "> No quests active.";
        }

        return null;
    }
}

File: src/main/kotlin/com/canefe/story/StoryPlaceholderExpansion.kt
package com.canefe.story

import me.clip.placeholderapi.expansion.PlaceholderExpansion
import org.bukkit.entity.Player

class StoryPlaceholderExpansion(
    private val plugin: Story,
) : PlaceholderExpansion() {
    override fun getAuthor(): String = "canefe"

    override fun getIdentifier(): String = "story"

    override fun getVersion(): String = "1.0.0"

    override fun persist(): Boolean = true

    override fun canRegister(): Boolean = true

    override fun onPlaceholderRequest(
        player: Player?,
        params: String,
    ): String? {
        if (player == null) {
            return ""
        }

        when {
            params.equals("quest_title", ignoreCase = true) -> {
                return player.player?.let { plugin.playerManager.getQuestTitle(it) }
            }
            params.equals("quest_objective", ignoreCase = true) -> {
                return player.player?.let { plugin.playerManager.getQuestObjective(it) }
            }
        }

        return null
    }
}
